packages/react/src/products/hooks/__tests__/useProductDetails.test.tsx

```diff
@@ -1,75 +1,186 @@
+  resetProductDetails,
+} from '@farfetch/blackout-redux';
+import {
+  mockProduct,
+  mockProductId,
+  mockProductsState,
+} from 'tests/__fixtures__/products';
 import { renderHook } from '@testing-library/react';
+import { withStore } from '@farfetch/blackout-react/tests/helpers';
+import useProductDetails from '../useProductDetails';
 
 jest.mock('@farfetch/blackout-redux', () => ({
   ...jest.requireActual('@farfetch/blackout-redux'),
-  fetchProductDetails: jest.fn(() => ({ type: 'get' })),
+  fetchProductDetails: jest.fn(() => () => Promise.resolve()),
+  resetProductDetails: jest.fn(() => () => Promise.resolve()),
 }));
 
-const getRenderedHook = (props, state = mockProductsState) => {
-  return renderHook(() => useProductDetails(props), {
-    wrapper: ({ children }) => (
-      <Provider store={mockStore(state)}>{children}</Provider>
-    ),
+describe('useProductDetails', () => {
+  it('should return data correctly with initial state', () => {
+    const { result } = renderHook(() => useProductDetails(mockProductId), {
+      wrapper: withStore(mockProductsState),
+    });
+
+    expect(result.current).toStrictEqual({
+      error: null,
+      isFetched: true,
+      isLoading: false,
+      data: {
+        ...mockProduct,
+        isOneSize: false,
+        isOutOfStock: false,
+      },
+      actions: {
+        reset: expect.any(Function),
+        refetch: expect.any(Function),
+      },
+    });
   });
-};
 
-describe('useProductDetails', () => {
-  beforeEach(jest.clearAllMocks);
-
-  describe('when it has a product id', () => {
-    it('should return values correctly with initial state', async () => {
-      const { result } = getRenderedHook(mockProductId);
-
-      expect(result.current).toStrictEqual({
-        breadcrumbs: expect.any(Array),
-        error: expect.any(Object),
-        groupedEntries: undefined,
-        isDuplicated: expect.any(Boolean),
-        isFetched: expect.any(Boolean),
-        isHydrated: expect.any(Boolean),
-        isInBag: expect.any(Boolean),
-        isLoading: expect.any(Boolean),
-        isOneSize: expect.any(Boolean),
-        isOutOfStock: expect.any(Boolean),
-        labelsPrioritized: expect.any(Array),
-        product: expect.any(Object),
-        promotions: expect.any(Array),
-        availableSizes: expect.any(Array),
-      });
+  it('should return error state', () => {
+    const errorMockProductsState = {
+      entities: {},
+      products: {
+        details: {
+          error: {
+            [mockProductId]: 'Error - Not loaded.',
+          },
+          isHydrated: {
+            [mockProductId]: false,
+          },
+          isLoading: {
+            [mockProductId]: false,
+          },
+        },
+      },
+    };
+
+    const { result } = renderHook(() => useProductDetails(mockProductId), {
+      wrapper: withStore(errorMockProductsState),
+    });
+
+    expect(result.current).toStrictEqual({
+      error: 'Error - Not loaded.',
+      isFetched: true,
```

Comment:
I think this should be false. I understand that when isFetched is true, we have product data. Is that something that our isProductFetched selector is missing? (This is on the correct line now :) )

Reply:
As discussed offline, we will change the way the `isFetched` prop works in all hooks to make it work like the `isFetched` value in react-query as much as possible. There was a bug in the details reducer that prevented this to work so a fix was introduced in another [PR](https://github.com/Farfetch/blackout/pull/431). Closing it...